queue.c

```diff
@@ -165,6 +165,10 @@ typedef xQUEUE Queue_t;
 /* The queue registry is simply an array of QueueRegistryItem_t structures.
  * The pcQueueName member of a structure being NULL is indicative of the
  * array position being vacant. */
+
+/* MISRA Ref 8.4.2 [Declaration shall be visible] */
+/* More details at: https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/MISRA.md#rule-84 */
+/* coverity[misra_c_2012_rule_8_4_violation] */
     PRIVILEGED_DATA QueueRegistryItem_t xQueueRegistry[ configQUEUE_REGISTRY_SIZE ];
```

Comment:
If ``xQueueRegistry`` is only needed to be visible for unit tests why not write this as

```c
#ifndef WHAT_EVER_THE_UNIT_TEST_MACRO_IS
static
#endif
PRIVILEGED_DATA QueueRegistryItem_t xQueueRegistry[ configQUEUE_REGISTRY_SIZE ];
```

then the visibility would be correct for both use cases.

My personal opinion in these *I need visibility for the unit test* cases is to rather have a ``ifdef/ifndef`` than an unnecessary visibility.